File: samples/PHP/bitwise-operations.php
function bitwise($a, $b)
{
    echo '$a AND $b: ', $a & $b, "\n";
    echo '$a OR $b: ', $a | $b, "\n";
    echo '$a XOR $b: ', $a ^ $b, "\n";
    echo 'NOT $a: ', ~$a, "\n";
    echo '$a << $b: ', $a << $b, "\n"; // left shift
    echo '$a >> $b: ', $a >> $b, "\n"; // arithmetic right shift
}

File: samples/Haskell/bitwise-operations.hs
import Data.Bits

bitwise :: Int -> Int -> IO ()
bitwise a b = do
  print $ a .&. b
  print $ a .|. b
  print $ a `xor` b
  print $ complement a
  print $ shiftL a b -- left shift
  print $ shiftR a b -- arithmetic right shift
  print $ shift a b  -- You can also use the "unified" shift function; positive is for left shift, negative is for right shift
  print $ shift a (-b)
  print $ rotateL a b -- rotate left
  print $ rotateR a b -- rotate right
  print $ rotate a b  -- You can also use the "unified" rotate function; positive is for left rotate, negative is for right rotate
  print $ rotate a (-b)

main = bitwise 255 170
